### PR TITLE
SAK-44151 General small improvements and fixes for Lessons

### PR DESCRIPTION
**Pull request by davidpbauer**
These improvements and bug fixes include:
- toggling a css class on section headers based on if the section is open or closed
- Fixing a typo in the polls markup causing styling inconsistencies
- Lowering all of the z-index values for lessons dialogs
- Fixing styling inconsistencies in Checklist feature
- Adding tabindex to the Learning Apps iframe
- Linked checklist items are broken when imported into a new site, but are not
indicated as broken to the user
- Change checklist item duplication to use JSON instead of String manipulation
- Fix date picker when no date is set
- Fix issue with checklist item indexes
- Fix auto select multiple choice on add question load
- Keep subpage group release info on duplication / import
- Fix group releasing LTI items in Lessons

### CODE REVIEW

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/LessonBuilderEntityProducer.java [hunk 1]
```diff
@@ -1006,31 +1033,41 @@ else if (type == SimplePageItem.ASSESSMENT)
...
+	Site oldSite = null;
+	try {
+		oldSite = siteService.getSite(fromSiteId);
+	} catch (Exception e) {
+		log.debug("site " + fromSiteId + " not found.", e);
+	}
+	// For each old group, get the corresponding new group id
+	if (!groups.isEmpty() && siteGroups != null && oldSite != null) {
+		final String[] parts = groups.split(",");
+		for (int n = 0; n < parts.length; n ++) {
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/LessonBuilderEntityProducer.java:1054]: I'd just use for (String part : parts) { here. Clearer, unless you need to test the index.

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/LessonBuilderEntityProducer.java [hunk 2]
```diff
@@ -510,6 +515,28 @@ else if (item.getType() == SimplePageItem.ASSESSMENT)
 		    addAttr(doc, itemElement, "samewindow", item.isSameWindow() ? "true" : "false");
 		
 		String attrString = item.getAttributeString(); //json encoded attributes
+
+		final JSONParser jsonParser = new JSONParser(); // JSON parser used to parse the attribute string into a JSON object
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/LessonBuilderEntityProducer.java:519]: Why not use j2ee JsonObject here, instead? I know simplepagebean uses Google's lib, but the j2ee version is a very similar api, with perhaps better builder patterns.

**Review by adrianfish — CHANGES_REQUESTED** [review]
Small stuff. Is this what you would call one of the "non contentious" tickets? The JIRA could do with a test plan, so your changes can be exercised. I know it's a feature, but particular parts of the UI that should be checked would be useful.

**Review by adrianfish — APPROVED** [review]

**Comment by adrianfish** [comment]
I've approved this. I pulled the branch and tested it out, and it all seems to work as advertised. You do need a test plan on the JIRA though, so QA can use it in their test scripts.

**Comment by davidpbauer** [comment]
Thanks, Adrian. Yes, I wanted to get the pull requests in as quickly as possible, so their JIRAs don't have much right now. I'll be filling in details and test plans this morning.

**Comment by adrianfish** [comment]
@davidpbauer Really appreciate you getting on this so fast.

**Comment by davidpbauer** [comment]
I've added some more details and test plans to SAK-44151. I can continue to provide more detail if needed.

**Pull request merged**